### PR TITLE
Support auditing event subscriptions

### PR DESCRIPTION
**Pull request by welteki**
## Description

Support configuration for auditing event subscriptions through the helm Chart.

## Why is this needed?

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Who is this for?

What company is this for? Are you listed in the [ADOPTERS.md](https://github.com/openfaas/faas/blob/master/ADOPTERS.md) file?

## How Has This Been Tested?

This chart has been used to verify the different configuration options while testing the auditing feature E2E on my local cluster.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: chart/openfaas/README.md [hunk 1]
```diff
@@ -607,6 +607,9 @@ yaml) |
 | `eventSubscription.insecureTLS` | Enable insecure TLS for webhook invocations | `false` |
 | `eventSubscription.metering.enabled` | Enable metering events. | `false` |
 | `eventSubscription.metering.defaultRAM` | Default memory value used in function_usage events for metering when no memory limit is set on the function.  | `40Mi` |
+| `eventSubscription.metering.excludedNamespaces` | Comma-separated list of namespaces to exclude from metering for when functions are used to handle the metering webhook events | `""` |
+| `eventSubscription.auditing.enabled` | Enable auditing events | `false` |
+| `eventSubscription.auditing.httpVerbs` | Comma-separated list of HTTP methods to audit | `"GET,PUT,POST,DELETE"`|
```
Comment (alexellis) [inline chart/openfaas/README.md:612]: Remove GET from the default list

File: chart/openfaas/README.md [hunk 2]
```diff
@@ -607,6 +607,9 @@ yaml) |
 | `eventSubscription.insecureTLS` | Enable insecure TLS for webhook invocations | `false` |
 | `eventSubscription.metering.enabled` | Enable metering events. | `false` |
 | `eventSubscription.metering.defaultRAM` | Default memory value used in function_usage events for metering when no memory limit is set on the function.  | `40Mi` |
+| `eventSubscription.metering.excludedNamespaces` | Comma-separated list of namespaces to exclude from metering for when functions are used to handle the metering webhook events | `""` |
```
Comment (alexellis) [inline chart/openfaas/README.md:610]: Thanks for adding that.

File: chart/openfaas/README.md [hunk 3]
```diff
@@ -607,6 +607,9 @@ yaml) |
 | `eventSubscription.insecureTLS` | Enable insecure TLS for webhook invocations | `false` |
 | `eventSubscription.metering.enabled` | Enable metering events. | `false` |
 | `eventSubscription.metering.defaultRAM` | Default memory value used in function_usage events for metering when no memory limit is set on the function.  | `40Mi` |
```
Comment (alexellis) [inline chart/openfaas/README.md:609]: This is 512Mi now

File: chart/openfaas/templates/gateway-dep.yaml [hunk 4]
```diff
@@ -309,6 +309,23 @@ spec:
...
+            value: "{{ .Values.nats.external.host }}"
+          - name: nats_port
+            value: "{{ .Values.nats.external.port }}"
+          {{- else }}
+          - name: nats_address
+            value: "nats.{{ .Release.Namespace }}.svc.{{ .Values.kubernetesDNSDomain }}"
+          - name: nats_port
+            value: "4222"
+          {{- end }}
+          - name: auditing
```
Comment (alexellis) [inline chart/openfaas/templates/gateway-dep.yaml:324]: How about only emit this whole section if and only if `.Values.eventSubscription.auditing.enabled`?

I did the same for metering to minimise the YAML on deployments for support purposes.

**Comment by alexellis** [comment]
Why are there two auditing env-vars on the gateway template?

File: chart/openfaas/templates/gateway-dep.yaml (hunk 4 above)
Comment (welteki) [reply chart/openfaas/templates/gateway-dep.yaml:324]: This is already in place, the whole configuration section for NATS and auditing is only included if `.Values.eventSubscription.auditing.enabled`

File: chart/openfaas/README.md (hunk 3 above)
Comment (welteki) [reply chart/openfaas/README.md]: This has been updated.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**